modules/client/src/tokenVoting/internal/client/methods.ts

```diff
@@ -738,4 +749,49 @@ export class TokenVotingClientMethods extends ClientCore
     }
     return null;
   }
+
+  /**
+   * Checks if the given token is compatible with the TokenVoting plugin
+   *
+   * @param {string} tokenAddress
+   * @return {*}  {Promise<TokenVotingTokenCompatibility>}
+   * @memberof TokenVotingClientMethods
+   */
+  public async isTokenVotingCompatibleToken(
+    tokenAddress: string,
+  ): Promise<TokenVotingTokenCompatibility> {
+    const signer = this.web3.getConnectedSigner();
+    // check if is address
+    if (!isAddress(tokenAddress) || tokenAddress === AddressZero) {
+      throw new InvalidAddressError();
+    }
+    const provider = this.web3.getProvider();
+    // check if is a contract
+    if (await provider.getCode(tokenAddress) === "0x") {
+      throw new NotAContractError();
+    }
+    const contract = new Contract(
+      tokenAddress,
+      ERC165_ABI,
+      signer,
+    );
+
+    if (!await isERC20Token(tokenAddress, signer)) {
+      return TokenVotingTokenCompatibility.INCOMPATIBLE;
+    }
+    try {
+      if (!await contract.supportsInterface(ERC165_INTERFACE_ID)) {
+        return TokenVotingTokenCompatibility.NEEDS_WRAP;
+      }
+      for (const iface of GOVERNANCE_INTERFACES_SUPPORTED) {
+        const isSupported = await contract.supportsInterface(iface);
+        if (isSupported) {
+          return TokenVotingTokenCompatibility.COMPATIBLE;
+        }
+      }
```

Comment:
This doesn't match with the contract
```solidity
if (
    // If token supports none of them
    // it's simply ERC20 which gets checked by _isERC20
    // Currently, not a satisfiable check.
    (!supportedIds[0] && !supportedIds[1] && !supportedIds[2]) ||
    // If token supports IERC20, but neither
    // IVotes nor IGovernanceWrappedERC20, it needs wrapping.
    (supportedIds[0] && !supportedIds[1] && !supportedIds[2])
) {
    token = governanceWrappedERC20Base.clone();
    // User already has a token. We need to wrap it in
    // GovernanceWrappedERC20 in order to make the token
    // include governance functionality.
    GovernanceWrappedERC20(token).initialize(
        IERC20Upgradeable(tokenSettings.addr),
        tokenSettings.name,
        tokenSettings.symbol
    );
}
```
because of the logic we have here, it doesn't actually matter if the ERC165 check for IERC20 (`supportedIds[0]`) is true or false. 🙃

It could be simplified to `if(!supportedIds[1] && !supportedIds[2]){...}` if I am not mistaken. 
The contract code is not good

Reply:
Ah true, maybe I was irritated by the test that was wrong.